src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [state];
+
+  for (let i = 0; i < actions.length; i++) {
+    const previousStateCopy = { ...states[i] };
+
+    switch (actions[i].type) {
```

Comment:
You can destructure all the needed props from current action (type, extraData, keysToRemove)